README.md

```diff
@@ -1,133 +1,414 @@
 
+  Closes connection from PostgreSQL to SQLite in the current session.
+
+- bool **sqlite_fdw_disconnect_all()**
+
+- SETOF record **sqlite_fdw_version()**;
+
+Identifier case handling
+------------------------
+
+PostgreSQL folds identifiers to lower case by default, SQlite is case insensetive by default. It's important
+to be aware of potential issues with table and column names.
+
+This SQL isn't correct for SQLite: `Error: duplicate column name: a`, but is correct for PostgreSQL
+
+```sql
+	CREATE TABLE T (
+		"A" INTEGER,
+		"a" NUMERIC
+	);
+```
+For SQLite there is no difference between
+
+```sql
+	select * from t;
+	select * from T;
+	select * from "t";
+	select * from "T";
+```
+For PostgreSQL `select * from "T";` is independend query to table `T`, not to table `t` as first queries.
+
+If there is
+
+```sql
+	CREATE TABLE T (
+		A INTEGER,
+		b REAL
+	);
+```
+in SQLite, both `a` and `A` , `b` and `B` columns will have the same real datasource in SQlite
+
+```sql
+	CREATE FOREIGN TABLE "SQLite test" (
+		"A" int4 NULL,
+		"B" float8 NULL,
+		"a" int8 NULL,
+		"b" numeric NULL
+	)
+	SERVER sqlite_server
+	OPTIONS (table 'T');
+```
+
+Generated columns
+-----------------
+
+SQLite provides support for [generated columns](https://www.sqlite.org/gencol.html).
+Behaviour of `sqlite_fdw` with this columns _isn't yet described_.
+
+Note that while `sqlite_fdw` will `insert` or `update` the generated column value
+in SQLite, there is nothing to stop the value being modified within SQLite,
+and hence no guarantee that in subsequent `SELECT` operations the column will
+still contain the expected generated value. This limitation also applies to
+`postgres_fdw`.
+
+For more details on generated columns see:
+
+- [Generated Columns](https://www.postgresql.org/docs/current/ddl-generated-columns.html)
+- [CREATE FOREIGN TABLE](https://www.postgresql.org/docs/current/sql-createforeigntable.html)
+
+Character set handling
+----------------------
+
+When `sqlite_fdw` connects to a SQLite [no character set metadata](https://www.sqlite.org/search?s=d&q=character+set)
+stored in SQLite. There is only [`PRAGMA encoding;`](https://www.sqlite.org/pragma.html#pragma_encoding) with UTF-only values (`UTF-8`, `UTF-16`, `UTF-16le`, `UTF-16be`). All strings are interpreted acording the PostgreSQL database's server encoding. It's not a problem
+if both PostgreSQL database and SQLite character data from database file has UTF-8 or UTF-16 encoding. Otherewise
+character interpretation transformation problems will occur.
+
+Character case functions such as `upper`, `lower` and other are not pushed down because they does not work with UNICODE character in SQLite.
+
+Examples
+--------
+
+### Install the extension:
+
+```sql
+    CREATE EXTENSION sqlite_fdw;
+```
+
+### Create a foreign server with appropriate configuration:
+
+Please specify SQLite database path using `database` option.
+
+```sql
+    CREATE SERVER sqlite_server
+      FOREIGN DATA WRAPPER sqlite_fdw
```

Comment:
Indent of SQLs are not unified.  
Please confirm and correct totally.

Reply:
Ok. Base indent \t, additional 2 \s. Look like nothing have forgotten.

Reply:
It is still inconsistent. 

For example, 1. "CREATE TABLE ... " is 2 indent level. But "CREATE SERVER ..." is 4 indent level.

For example, 2. SELECT statement in "Access foreign table" consists of 2 lines. But that of "Identifier case handling" is only 1 line.

Reply:
SELECTs fixed.

Reply:
> CREATE TABLE ... " is 2 indent level. But "CREATE SERVER ..." is 4 indent level.

Outdated version? All SQL have got initial \t. Please comment on the line again. I think near https://github.com/mkgrgis/sqlite_fdw/blame/patch-1/README.md#L296 fixed, but maybe I misunderstand your comment, @t-kataym .

Reply:
Please let me explain.

There are 2 type pf styles.
Style1:
```
	CREATE SERVER sqlite_server --- Level1
	  FOREIGN DATA WRAPPER sqlite_fdw --- Level2
	  OPTIONS ( --- Level2
            database '/path/to/database' --- Level3
	  ); --- Level3
```

Style2:
```
	CREATE FOREIGN TABLE t1 ( --- Level1
	  a integer, --- Level2
	  b text) --- Level2
	SERVER sqlite_server --- Level1
	OPTIONS ( --- Level1
	  table 't1_sqlite' --- Level2
	); --- Level1
```

Because firebird_fdw uses Style2, please fix Style1 to follow Style2.

About SELECT statement, firebird_fdw describes by 1 line. So please follow it also.

Reply:
Thank @t-kataym ! There is no misunderstanding now. Also I have simplified `SELECT ... FROM` to 1 line by example in `firebird_fdw`. Yes, DDL and DSL style can be different. I think there is no indent problems in https://github.com/mkgrgis/sqlite_fdw/commit/66fe643910593cb224d13db925cef176f8620808 README.md. Otherwise feel free to reopen review of my commits.